src/libraries/Common/tests/System/Xml/XmlDiff/XmlDiff.cs

```diff
@@ -437,6 +438,20 @@ private DiffType CompareNodes(XmlDiffNode sourceNode, XmlDiffNode targetNode)
             return DiffType.Success;
         }
 
+        private string Normalize(string value)
+        {
+            if (String.IsNullOrEmpty(value))
+                return value;
+
+            if (NormalizeSpaces)
+                value = Regex.Replace(value, "[\u00A0\u180E\u2000-\u200B\u202F\u205F\u3000\uFEFF]", " ");
```

Comment:
>"[\u00A0\u180E\u2000-\u200B\u202F\u205F\u3000\uFEFF]" [](http://example.com/codeflow?start=45&length=53)

From where did you get this list? 

`\u180E` is a format character and not really a space. `\uFEFF` similar. And missing `\u1680`.

[https://unicodeplus.com/category/Zs](https://unicodeplus.com/category/Zs)

Reply:
The list I used is [Unicode spaces](https://jkorpela.fi/chars/spaces.html) pointed at by this [answer](https://stackoverflow.com/a/51399843/2076) and it is based on the [Unicode category Zs](https://www.compart.com/en/unicode/category/Zs).

It adds [\FEFF is zero-width no-break space](https://www.compart.com/en/unicode/U+FEFF) and seems to be missed by a lot of folks.

It adds [\u180E Mongolian vowel separator](https://www.compart.com/en/unicode/U+180E) because it is treated as a space, happy to remove it.

I removed [\u1680-Ogham space mark](https://www.compart.com/en/unicode/U+1680) because its glyph is usually rendered as a dash,  could easily put it back.

Happy to adjust this pattern as desired.

Reply:
It's not entirely about XML's view of spaces, this whole change is motivated because in the test-environment's locale the format for times is `h:mm AM` except the space _isn't actually a space_ but is really the [\u202F Narrow No-Break Space](https://unicodeplus.com/U+202F) which broke my naive test. When the need for a normalization to spaces (to defend against locale-driven possibilities) I figured I should cast a slightly wider net.
 
There's absolutely no objection in my mind to just using `\s` in the pattern since that only result in ignoring a few more 
"blanks" and this is nothing but test code anyway. Point me and I'll shoot as ordered :)

Reply:
In the date/time parsing code we just call [`char.IsWhiteSpace`](https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/Globalization/DateTimeParse.cs,5486). Either you continue using regex as you suggested or just manually create a helper method iterate over the string checking using `char.IsWhiteSpace`. What ever you feel easier to you, do it :-)

Thanks for your help fixing the tests!

Reply:
one note, if you use the regex, most likely you need `\p{Zs}` instead of `\s`. I don't recall if `\s` cover all different whitespaces or not.